system/jlib/jfile.cpp

```diff
@@ -2410,6 +2445,57 @@ IFileAsyncResult *CFileAsyncIO::writeAsync(offset_t pos, size32_t len, const voi
 
 #endif
 
+//---------------------------------------------------------------------------
+
+void FileIOStats::merge(IFileIO * io)
+{
+    if (!io)
+        return;
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        values[stat] += io->getStatistic((FileIOStat)stat);
+}
+
+void FileIOStats::merge(const FileIOStats & other)
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        values[stat] += other.values[stat];
+}
+
+void FileIOStats::merge(MemoryBuffer & in)
+{
+    unsigned __int64 value;
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+    {
+        in.read(value);
+        values[stat] += value;
+    }
+}
+
+void FileIOStats::deserialize(MemoryBuffer & in)
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        in.read(values[stat]);
+}
+
+#if 0
+void FileIOStats::serialize(MemoryBuffer & out) const
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        out.append(values[stat]);
+}
+#endif
+
+void FileIOStats::serializeNormalized(MemoryBuffer & out) const
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+    {
+        unsigned __int64 curValue = values[stat];
+        if (stat == FIOSiocycles)
+            curValue = cycle_to_nanosec(curValue);
```

Comment:
I can probably see why, other timing stats are converted at serialization time..

Perhaps we should serialize cycles and only convert when getStats() is processed.
Or if we're going to leave as serializing NS, then perhaps should classify the stat type here (in FileIOStats), instead of:
```
if (stat == FIOSiocycles)
```
. something like:
```
if (statType_cycleToNs == (stat & statTypeMask))
```